script/eval.go

```diff
@@ -57,21 +57,56 @@ func evalSnapshot(r *runner, n node) error {
 	return nil
 }
 
+// override images with new one for services under a tenant or only for a particular service if it is specified
+// possible usages:
+// set image for all services under top level tenant
+//SVC_USE <new image>
+//override old images with new for all services under top level tenant
+//SVC_USE <new image> <image to replace> replace image with new for all services under top level tenant
+// override old images with new for a specific service
+// SVC_USE <new image> <image to replace> service Zenoss.resmgr/Infrastructure/mariadb-model
 func evalUSE(r *runner, n node) error {
 
 	imageName := n.args[0]
-	replaceImgs := make([]string, len(n.args)-1)
-	//if len(n.args) > 1 && n.args[1] != "" {
+	replaceImgs := make([]string, 0)
+	var svcPath, serviceID string
+
+	tenantID, found := r.env["TENANT_ID"]
+	if !found {
+		return fmt.Errorf("no service tenant id specified for %s", USE)
+	}
+
 	if len(n.args) > 1 {
-		replaceImgs = n.args[1:]
+		for i, arg := range n.args[1:] {
+			if arg == "service" {
```

Comment:
I forgot to save this comment yesterday.  What happens when the keyword "service" is in the wrong location?\
eg. SVC_USE <new image> service <image to replace>  Zenoss.resmgr/Infrastructure/mariadb-model

Would like a test case for that.

Reply:
updated parser,  added check whether we have image to replace in the command and check if we have path_to_service specified.